modules/serverpod_auth/serverpod_auth_server/lib/src/business/user_authentication.dart

```diff
@@ -34,21 +45,73 @@ class UserAuthentication {
       method: method,
     );
 
-    session.updateAuthenticated(AuthenticationInfo(userId, scopes));
     var result = await AuthKey.db.insertRow(session, authKey);
+
+    if (updateSession) {
+      session.updateAuthenticated(
+        AuthenticationInfo(
+          userId,
+          scopes,
+          authId: '${result.id}',
+        ),
+      );
+    }
     return result.copyWith(key: key);
```

Comment:
Question: Was the copyWith still required here?

Reply:
nope ur right.